brave-cxf3/src/main/java/com/github/kristofa/brave/cxf3/TracerContext.java

```diff
@@ -0,0 +1,67 @@
+package com.github.kristofa.brave.cxf3;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ServerSpan;
+import com.github.kristofa.brave.ServerSpanThreadBinder;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Used to handle {@link ServerSpan} retrieved from cxf message exchange.
+ */
+public class TracerContext {
+
+  private final ServerSpanThreadBinder serverSpanThreadBinder;
+  private final ServerSpan serverSpan;
+
+  TracerContext(Brave brave, ServerSpan serverSpan) {
+    this.serverSpanThreadBinder = brave.serverSpanThreadBinder();
+    this.serverSpan = serverSpan;
+  }
+
+  /**
+   * Wraps provided {@link Runnable and ensures same {@link ServerSpan} context as the thread from which the Runnable was executed.
+   * {@link ServerSpan} is taken from cxf {@link org.apache.cxf.message.Message} {@link org.apache.cxf.message.Exchange}
+   *
+   * @param command {@link Runnable}
+   * @return wrapped {@link Runnable}
+   */
+  public Runnable wrap(Runnable command) {
+    if (command == null) {
+      return null;
+    }
+
+    // we should clean this thread, ServerSpan will finish on another one
+    serverSpanThreadBinder.setCurrentSpan(null);
+
+    return () -> {
+      if (serverSpan != null) {
+        serverSpanThreadBinder.setCurrentSpan(serverSpan);
+      }
+      command.run();
+    };
+  }
+
+  /**
+   * Wraps provided {@link Callable} and ensures same {@link ServerSpan} context as the thread from which the Callable was executed.
+   * {@link ServerSpan} is taken from cxf {@link org.apache.cxf.message.Message} {@link org.apache.cxf.message.Exchange}
+   *
+   * @param command {@link Callable}
+   * @return wrapped {@link Callable}
+   */
+  public <V> Callable<V> wrap(Callable<V> command) {
+    if (command == null) {
+      return null;
+    }
+
+    // we should clean this thread, ServerSpan will finish on another one
+    serverSpanThreadBinder.setCurrentSpan(null);
```

Comment:
@reta

I've separated jax-rs and jax-ws a little bit, they are still based on cxf interceptors.

To summarize what was changed as a whole:

* traceId is propagated
* is async scenarios
	* `TracerContext` must be used to propagate traceId to other thread
	* initial thread is cleaned
* cxf server features, separate for jax-ws and jax-rs
	* `BraveJaxRsServerFeature`
	* `BraveJaxWsServerFeature`
	* `cxf-rt-frontend-jaxws` and `cxf-rt-frontend-jaxrs` deps are in provided scope, if someone want to use only jax-ws, there's no dependency for jax-rs stuff.
* cxf feature for clients `BraveClientFeature`